src/stats/boxplot.jl

```diff
@@ -169,11 +168,35 @@ function Makie.plot!(plot::BoxPlot)
     t_segments = @lift($signals.t_segments)
     boxwidth = @lift($signals.boxwidth)
 
-    outliercolor = lift(plot[:outliercolor], plot[:color]) do outliercolor, color
-        outliercolor === automatic || return outliercolor
-        c = to_color(color)
-        return RGB(red(c), green(c), blue(c))
-    end
+    outliercolor = lift(plot[:outliercolor], plot[:color], centers, outliers, orientation) do outliercolor, color, centers, outliers, orientation
```

Comment:
I think this is very complex, because `signals` does not store enough information, so you end up playing around with some other attributes.

I imagine it would be easier to also return `outlier_indices` in `signals` (by storing in the output not only the value of the `outlier` but also its index within `plot[2]`). This should only require a small refactor to iterate not `value in view(y, idxs)` but rather `idx in idxs`.

Once you have access to that, it is much easier / cleaner to see what the corresponding color should be.

Reply:
I think the last commit should address this.